thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/AnomalyFunctionDTO.java

```diff
@@ -70,48 +68,10 @@ public boolean isToCalculateGlobalMetric() {
    * @return a Properties object corresponds to the properties String of this anomaly function.
    */
   public Properties toProperties() {
-    Properties props = null;
-    try {
-      props = toProperties(getProperties());
-    } catch (IOException e) {
-      LOGGER.warn("Failed to parse property string ({}) for anomaly function: {}", getProperties(), getId());
-
-    }
+    Properties props = StringUtils.decodeCompactedProperties(getProperties());
     return props;
   }
 
-  /**
-   * Parses the properties of String and returns the corresponding Properties object.
-   *
-   * @return a Properties object corresponds to the properties String of this anomaly function.
-   */
-  public static Properties toProperties(String properties) throws IOException {
```

Comment:
This is a public method. Are you sure that it is not called by some other code?

Reply:
Thanks for pointing out.

Reply:
Added back.